src/LoginForm/LoginForm.test.js

```diff
@@ -24,24 +24,21 @@ describe("LoginForm", () => {
   it("returns form information to callback when successfully validated", async () => {
     const onLogin = jest.fn(data => data);
     const elements = setup({ onLogin });
-    await act(async () => {
-      userEvent.type(elements.inputs.email, "joe.bloggs@domain.com");
-      userEvent.type(elements.inputs.password, "indigo shark wallplug");
-      fireEvent.submit(elements.submit);
-    });
-    expect(onLogin).toHaveReturnedWith({
-      email: "joe.bloggs@domain.com",
-      password: "indigo shark wallplug"
-    });
+    userEvent.type(elements.inputs.email, "joe.bloggs@domain.com");
+    userEvent.type(elements.inputs.password, "indigo shark wallplug");
+    fireEvent.submit(elements.submit);
+    await waitFor(() =>
+      expect(onLogin).toHaveReturnedWith({
+        email: "joe.bloggs@domain.com",
+        password: "indigo shark wallplug"
+      })
+    );
   });
   it("doesnt call callback on validation errors", async () => {
     const onLogin = jest.fn();
     const elements = setup({ onLogin });
-    await act(async () => {
-      userEvent.type(elements.inputs.email, "joe.bloggs");
-      // missing password
-      fireEvent.submit(elements.submit);
-    });
-    expect(onLogin).not.toHaveBeenCalled();
+    userEvent.type(elements.inputs.email, "joe.bloggs"); // missing password
+    fireEvent.submit(elements.submit);
+    await waitFor(() => expect(onLogin).not.toHaveBeenCalled());
```

Comment:
This should now be testing for the error message on screen as we discovered. onLogin is not called by default.